clang/test/CodeGenCXX/constexpr-math.cpp

```diff
@@ -0,0 +1,106 @@
+// CHECK-NEXT:    store double 1.300000e+00, ptr [[MIN1]], align 8
+// CHECK-NEXT:    store double -0.000000e+00, ptr [[MIN2]], align 8
+// CHECK-NEXT:    store double -0.000000e+00, ptr [[MIN3]], align 8
+// CHECK-NEXT:    store float 0x7FF8000000000000, ptr [[MIN4]], align 4
+// CHECK-NEXT:    store float -1.000000e+00, ptr [[MIN5]], align 4
+// CHECK-NEXT:    store float 0xFFF0000000000000, ptr [[MIN6]], align 4
+// CHECK-NEXT:    store float 0.000000e+00, ptr [[MIN7]], align 4
+// CHECK-NEXT:    store x86_fp80 0xK4005F6E978D4FDF3B646, ptr [[MIN8]], align 16
+// CHECK-NEXT:    store double 1.524000e+01, ptr [[MAX1]], align 8
+// CHECK-NEXT:    store double 0.000000e+00, ptr [[MAX2]], align 8
+// CHECK-NEXT:    store double 0.000000e+00, ptr [[MAX3]], align 8
+// CHECK-NEXT:    store float -1.000000e+00, ptr [[MAX4]], align 4
+// CHECK-NEXT:    store float 0x7FF0000000000000, ptr [[MAX5]], align 4
+// CHECK-NEXT:    store float 0.000000e+00, ptr [[MAX6]], align 4
+// CHECK-NEXT:    store float 0x7FF8000000000000, ptr [[MAX7]], align 4
+// CHECK-NEXT:    store x86_fp80 0xK4008C540C49BA5E353F8, ptr [[MAX8]], align 16
+// CHECK-NEXT:    store double 0x3FEEDCCCCCCCCCCD, ptr [[FREXP1]], align 8
+// CHECK-NEXT:    store double 0.000000e+00, ptr [[FREXP2]], align 8
+// CHECK-NEXT:    store double -0.000000e+00, ptr [[FREXP3]], align 8
+// CHECK-NEXT:    store float 0x7FF8000000000000, ptr [[FREXP4]], align 4
+// CHECK-NEXT:    store float 0xFFF8000000000000, ptr [[FREXP5]], align 4
+// CHECK-NEXT:    store float 0x7FF0000000000000, ptr [[FREXP6]], align 4
+// CHECK-NEXT:    store float 0xFFF0000000000000, ptr [[FREXP7]], align 4
+// CHECK-NEXT:    store x86_fp80 0xK3FFE81A9FBE76C8B4396, ptr [[FREXP8]], align 16
+// CHECK-NEXT:    store float 8.750000e-01, ptr [[FREXP9]], align 4
+// CHECK-NEXT:    store x86_fp80 0xK3FFEEAC7AE147AE14800, ptr [[FREXP10]], align 16
+// CHECK-NEXT:    ret i32 0
+//
+int func()
+{
+  int i;
+
+  // fmin
+  constexpr double min1 = __builtin_fmin(15.24, 1.3);
+  constexpr double min2 = __builtin_fmin(-0.0, +0.0);
+  constexpr double min3 = __builtin_fmin(+0.0, -0.0);
+  constexpr float min4 = __builtin_fminf(NAN, NAN);
+  constexpr float min5 = __builtin_fminf(NAN, -1);
+  constexpr float min6 = __builtin_fminf(-INFINITY, 0);
+  constexpr float min7 = __builtin_fminf(INFINITY, 0);
+  constexpr long double min8 = __builtin_fminl(123.456L, 789.012L);
+
+  // fmax
+  constexpr double max1 =  __builtin_fmax(15.24, 1.3);
+  constexpr double max2 = __builtin_fmax(-0.0, +0.0);
+  constexpr double max3 = __builtin_fmax(+0.0, -0.0);
+  constexpr float max4 = __builtin_fmaxf(NAN, -1);
+  constexpr float max5 = __builtin_fmaxf(+INFINITY, 0);
+  constexpr float max6 = __builtin_fmaxf(-INFINITY, 0);
+  constexpr float max7 = __builtin_fmaxf(NAN, NAN);
+  constexpr long double max8 = __builtin_fmaxl(123.456L, 789.012L);
+
+  // frexp
+  constexpr double frexp1 = __builtin_frexp(123.45, &i);
```

Comment:
@zahiraam, to be explicit, these tests (as-is) will fail once the comment re: `StoreExponent()` is addressed.

This will probably work:
```suggestion
  constexpr double frexp1 = __builtin_frexp(123.45, (int [1]){});
```

Reply:
Done.